main/main.c

```diff
@@ -102,6 +140,72 @@ rte_strdup(const char *type, const char *s)
 	return strcpy(res, s);
 }
 
+FILE *log_file = NULL;
+struct rte_timer log_timer;
+
+static int
+cleanup_log(void)
+{
+	if (log_file != NULL) {
+		fclose(log_file);
+		log_file = NULL;
+	}
+
+	rte_free(log_base_dir);
+	rte_free(log_file_name_format);
+
+	return 0;
+}
+
+int
+gatekeeper_log_init(void)
+{
+	int ret;
+	time_t now;
+	struct tm *time_info;
+	char log_file_name[100];
+	char log_file_path[100];
```

Comment:
Let's be generous here since the values come from users, so use 128 for `log_file_name` and 512 for `log_file_path`.